### ISSUE TITLE
Items Ending up in Different Building Inventories

### ISSUE THREAD

**Issue by Enverex** [ldtteam/minecolonies#2877]
### Minecolonies version
Version: 1.12.2-0.9.126-ALPHA

### Expected behavior
Items to be taken to the warehouse.

### Actual behaviour
Items sometimes end up in other workers building inventory.

### Steps to reproduce the problem
Unknown, this occurred in a decent sized town though.

This is a picture of my fisherman's hut inventory. Note the seeds and wheat (presumably from the farmer). That said, the doors and diorite must have come from either the warehouse or the builders hut, so it's as if those have been passed around for some reason.

![wronginventory](https://user-images.githubusercontent.com/958670/45598358-b07c3180-b9d2-11e8-922d-0992d11671ce.png)

My warehouse did seem to be getting near max capacity though so is it possible that things started to go a little weird once it reached that stage?

**Comment by Raycoms** [comment]
Did the deliveryman complain about a full warehouse?

Am So., 16. Sep. 2018 um 13:06 Uhr schrieb Benjamin Hodgetts <
notifications@github.com>:

**Comment by Enverex** [comment]
I've not seen that message at any point, but it could have happened when my client wasn't connected.

**Comment by Enverex** [comment]
No messages about the warehouse being full and it's no-where near full anymore, but random things still seem to be getting delivered to my newly built (second) builders hut.

![image](https://user-images.githubusercontent.com/958670/45599865-1d032a80-b9eb-11e8-82fd-2367ee4f55de.png)

Wool from the shepherd that you can see in the background and wheat from a farm that's a decent distance away. Then a misc selection of items where I have no idea their origin.

**Comment by MineShadow7** [comment]
I think there was a barbarian raid that killed both of your builder and farmer. Then the fisherman came to his work and picked up the items on the ground.

**Comment by Enverex** [comment]
There's never been a barbarian raid and the server is currently set to peaceful. The builder and farmer haven't been killed. I built this builder hut and these items appeared all within a single session of me being online so I would have seen warehouse messages as well.

**Comment by Enverex** [comment]
My other build also appears to have an inventory of crap that A) shouldn't have been put there in the first place and B) isn't being emptied out. Items such as Wool, Wheat and Seeds. Runners/Deliverymen are now complaining they can't deliver needed items because the inventory is full, but that's only because it's full unnecessary items.

![image](https://user-images.githubusercontent.com/958670/45601065-7de82e00-b9fe-11e8-8b35-d84439b86706.png)

**Comment by Inomutso** [comment]
Hi there
we are running
 
Minecolonie Version  minecolonies-1.12.2-0.9.132-ALPHA-universal.jar
on our hosted Server

we have one colonie running and I can see the same issue here, as Enverex has reported

we have 6 Deliverymen runnig and only some of the workers building inventory´s, are being
filled up with a huge amount of items, sometimes two times in a MC day.

It´s not that easy to find out, who of our 6 DLmen is doing this. I found out, that two of them,
with their Hut´s in the same colonie area, where some of our workers hut´s are, do that.

so the fisherman (2) the composter´s (3) the smelter (1) the shepard (1)  are affected
and their chests are being filled up randomly. We have 2 Composter´s in the same area, 
as these two DLmen are usually working, but only one of them is affected.

for example one of our Miner´s (2) the cowboy (1) one farmer (2) one cook (2) in the same area
are not affected

I tried to fire these two DLmen and put in other workers for this job, but they still do the same.

Our Warehouse is about 50% full and there was no message about a full Warehouse.

I tried to fix it with /mc colony rs reset , but there was no change.

Deliverymen are still complaining they can't deliver needed items because the inventory is full,
but that's only because it's full of unnecessary items.

I can see this thince Version minecolonies-1.12.2-0.9.8-BETA-universal.jar, as far as I remember

we have our minecolonie thince Version minecolonies-universal-1.12.2-0.8.4932.jar, still loving it
